transport.go

```diff
@@ -61,11 +81,19 @@ func (t *Transport) SecureInbound(ctx context.Context, insecure net.Conn) (cs.Co
 // If the handshake fails, the server will close the connection. The client will
 // notice this after 1 RTT when calling Read.
 func (t *Transport) SecureOutbound(ctx context.Context, insecure net.Conn, p peer.ID) (cs.Conn, error) {
-	cl := tls.Client(insecure, t.identity.ConfigForPeer(p))
-	return t.handshake(ctx, cl)
+	verifiedCallback := func(pubKey ic.PubKey) {
+		t.activeMutex.Lock()
+		t.active[insecure] = pubKey
```

Comment:
Won't this leak the conn?

Reply:
It is, but only because I forgot the `defer` statement that cleans up the map, as in `SecureInbound`.